core/arch/x86/decode.c

```diff
@@ -1226,19 +1354,25 @@ decode_reg(decode_reg_t which_reg, decode_info_t *di, byte optype, opnd_size_t o
     case TYPE_V:
     case TYPE_W:
     case TYPE_V_MODRM:
-    case TYPE_VSIB:
-        return ((TEST(PREFIX_VEX_L, di->prefixes) &&
-                 /* Not only do we use this for .LIG (where raw reg is either
-                  * OPSZ_32 or OPSZ_16_vex32) but also for VSIB which currently
-                  * does not get up to OPSZ_16 so we can use this negative check.
-                  */
-                 expand_subreg_size(opsize) != OPSZ_16)
-                    ? (extend ? (REG_START_YMM + 8 + reg) : (REG_START_YMM + reg))
-                    : (extend ? (REG_START_XMM + 8 + reg) : (REG_START_XMM + reg)));
+    case TYPE_VSIB: {
+      byte extend_reg = extend ? reg + 8 : reg;
```

Comment:
I think there should be another additional notion of extended for the 16 new upper registers of AVX-512 in some form or the other, mentioned in previous comment. How are the upper 16 registers currently encoded?